### PR TITLE
Make TransactionInfo.txData an union

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/21078

Part of TxService refactoring for multi chain support.
Review by per commit might be easier.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by yrliou** [comment]
need to fix front-end test errors, not ready for review yet.

**Comment by yrliou** [comment]
This is currently blocked by adding union mojo binding support in iOS, cc @kylehickinson.

**Comment by yrliou** [comment]
Still have to wait for Kyle to add iOS support for union in order to merge, but other platforms are ready for review.

**Review by darkdh — APPROVED** [review]
++ for non-iOS part

File: build/ios/mojom/mojom_objc_generator.py [hunk 1]
```diff
@@ -409,13 +428,15 @@ def is_move_only_kind(kind):
...
-    def _GetObjCPropertyModifiers(self, kind):
-        modifiers = ['nonatomic']
+    def _GetObjCPropertyModifiers(self, kind, inside_union=False):
+        modifiers = {'nonatomic'}
         if (mojom.IsArrayKind(kind) or mojom.IsStringKind(kind) or
                 mojom.IsMapKind(kind) or mojom.IsStructKind(kind)):
-            modifiers.append('copy')
+            modifiers.add('copy')
+            if inside_union:
+                modifiers.add('nullable')
```
Comment (kylehickinson) [inline build/ios/mojom/mojom_objc_generator.py:437]: Realizing I need to add more times where `nullable` is added (nested union for instance)

**Review by SergeyZhukovsky — APPROVED** [review]
Android part looks good!

**Review by kylehickinson — APPROVED** [review]
Fixed union gen and nullable thing I caught, iOS++

**Review by Douglashdaniel — APPROVED** [review]
++

**Pull request merged**